autogen.pl

```diff
@@ -776,6 +776,43 @@ sub find_and_delete {
 # Find a specific executable and ensure that it is a recent enough
 # version.
 
+sub check_m4_gnu_extensions {
+    verbose "Checking whether m4 supports the gnu extensions\n";
+    my $m4 = $ENV{'M4'};
+    $m4 = "m4" unless (defined $m4);
+    my $result = `$m4 -g config/check_m4_gnu_extension.m4`;
+    if ($? != 0) {
+        $result = `$m4 config/check_m4_gnu_extension.m4`;
+    }
+    if ($result eq "ten\n") {
+        verbose "     ==> SUPPORTED\n";
+    } elsif ($result eq "one0\n") {
+        verbose "     ==> NOT SUPPORTED\n";
+        print "\nI am gonna abort.  :-(
+m4 must support GNU extensions
+
+if you are running on OpenBSD, you might want to
+export M4=gm4
+and run autogen.pl again
+
+=================================================================\n";
+        my_exit(1);
+    } else {
+        print "\nI am gonna abort.  :-(
+m4 must support GNU extensions
+$m4 -g config/check_m4_gnu_extension.m4 returned an unexpected result
+\"$result\", expected value is \"ten\", and known unexpected value is \"one0\"
+
+=================================================================\n";
+        my_exit(1);
```

Comment:
I don't know if you really need two different error messages here -- if you care, you could combine them into a single error message (I don't feel strongly; it is just slightly more code to support).